### PR TITLE
feat: ignore_message_queues configuration

### PR DESCRIPTION
**Pull request by astorm**
Partially implements: https://github.com/elastic/apm-agent-nodejs/issues/1956

This PR adds [the `ignore_message_queues`](https://github.com/elastic/apm/blob/master/specs/agents/tracing-instrumentation-messaging.md#ignore_message_queues-configuration) configuration to the agent.  This is a prerequisite for supporting AWS SQS instrumentation.  This PR does not include a CHANGELOG entry (that will come later, once the first messaging system (SQS) is added)

- [x] Implement code
- [X] Add tests
- [X] Update TypeScript typings
- [X] Update documentation
- [X] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/master/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -933,3 +933,33 @@ During startup the Node.js agent queries the local environment to determine whet
 - `none`: Do not query for any cloud provider information.
 
 If the value is not one of the five listed above, the agent will use the value of `auto`.
+
+[[ignore_message_queues]]
```
Comment (trentm) [inline docs/configuration.asciidoc:937]: Typically we use hyphens for the HTML anchors.

Aside: Do you have suggestion/ideas for where to include the actual string "ignore_message_queues" in changes like this? The Node.js Agent spells its config vars in the different ignoreMessageQueues pattern which breaks [the grep test](http://jamie-wong.com/2013/07/12/grep-test/). This need not block this change, however.

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -933,3 +933,33 @@ During startup the Node.js agent queries the local environment to determine whet
...
+* *Type:* Array
+* *Default:* `[]`
+* *Env:* `ELASTIC_IGNORE_MESSAGE_QUEUES`
+
+Array or comma-separated string of wildcard patterns that tell the agent to
+ignore certain queues/topics when instrumenting messaging systems.
+
+When an instrumented messaging system sends or receives a message, the agent
+will test the queue/topic name against each wildcard in this list. If
+the name matches the agent will skip instrumenting the operation.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:948]:
```suggestion
the name matches, the agent will skip instrumenting the operation.
```

File: docs/configuration.asciidoc (hunk 1 above)
Comment (bmorelli25) [inline docs/configuration.asciidoc:937]:
```suggestion
[[ignore-message-queues]]
```

How do you feel about dashes here to maintain consistency with the other anchors?

File: docs/configuration.asciidoc [hunk 3]
```diff
@@ -933,3 +933,33 @@ During startup the Node.js agent queries the local environment to determine whet
...
+
+Example usage:
+
+[source,js]
+----
+require('elastic-apm-node').start({
+  ignoreMessageQueues: [
+    'overnight_jobs',
+    'events_*',
+    '(?-i)/caseSensitiveSearch'
```
Comment (trentm) [inline docs/configuration.asciidoc:962]:
```suggestion
    '(?-i)caseSensitiveSearch'
```

Perhaps drop the slash... unless it is typical to have slashes in queue names. This example came from transactionIgnoreUrls where the slashes are less likely to be confused as being part of the syntax.

File: docs/configuration.asciidoc [hunk 4]
```diff
@@ -933,3 +933,33 @@ During startup the Node.js agent queries the local environment to determine whet
 - `none`: Do not query for any cloud provider information.
 
 If the value is not one of the five listed above, the agent will use the value of `auto`.
+
+[[ignore_message_queues]]
+==== `ignoreMessageQueues`
+* *Type:* Array
+* *Default:* `[]`
+* *Env:* `ELASTIC_IGNORE_MESSAGE_QUEUES`
+
```
Comment (trentm) [inline docs/configuration.asciidoc:942]: Are you adding support for central config later?

File: docs/configuration.asciidoc (hunk 1 above)
Comment (astorm) [reply docs/configuration.asciidoc:937]: 👍 -- making this change.  @bmorelli25 are there automated docs linting tools that would catch this sort of thing for us?

Comment (astorm) [reply docs/configuration.asciidoc:937]: 👍 switching to hyphens.  I don't have any good suggestions/ideas for where the literal text `ignore_message_queues` should go.  The decision to stick with camel casing binds our hands there.  I think the best we can do is the more recent convention of prepending `ELASTIC_` to get the env variable name (which presumes a `grep -i` test).

File: docs/configuration.asciidoc (hunk 4 above)
Comment (astorm) [reply docs/configuration.asciidoc:942]: Ah -- good catch.  I'd missed/assumed that this wasn't a configuration intended for central config, but re-reading the spec confirms it is.   

I've now added it to the `CENTRAL_CONFIG` array in `lib/config.js` and updated these docs and tested the configuration with a mock APM server.   I'll also need to cargo cult this PR in Kibana to make this configuration a central config. https://github.com/elastic/kibana/pull/85655

File: docs/configuration.asciidoc (hunk 2 above)
Comment (astorm) [reply docs/configuration.asciidoc:948]: ✅

File: docs/configuration.asciidoc (hunk 3 above)
Comment (astorm) [reply docs/configuration.asciidoc:962]: ✅ making this change

**Review by trentm — APPROVED** [review]

**Pull request merged**